app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -592,41 +573,26 @@ class NewMessageViewModel @Inject constructor(
         isTaskRoot: Boolean,
         startWorkerCallback: () -> Unit,
     ) = globalCoroutineScope.launch(ioDispatcher) {
-        autoSaveJob?.cancel()
 
-        if (isSavingDraftWithoutChanges(action)) {
-            if (isNewMessage) removeDraftFromRealm()
+        if (isFinishing && isSavingDraftWithoutChanges(action)) {
+            if (!arrivedFromExistingDraft) removeDraftFromRealm()
             return@launch
         }
 
         context.trackSendingDraftEvent(action, draft, currentMailbox.externalMailFlagEnabled)
 
         saveDraftToLocal(action)
-        showDraftToastToUser(action, isFinishing, isTaskRoot)
 
-        startWorkerCallback()
-        if (action == DraftAction.SAVE && !isFinishing) {
-            isNewMessage = false
-            saveDraftSnapshot()
+        if (isFinishing) {
+            if (isTaskRoot) showDraftToastToUser(action)
+            startWorkerCallback()
         }
     }
 
-    private suspend fun showDraftToastToUser(
-        action: DraftAction,
-        isFinishing: Boolean,
-        isTaskRoot: Boolean,
-    ) = withContext(mainDispatcher) {
+    private suspend fun showDraftToastToUser(action: DraftAction) = withContext(mainDispatcher) {
         when (action) {
-            DraftAction.SAVE -> {
-                if (isFinishing) {
-                    if (isTaskRoot) context.showToast(R.string.snackbarDraftSaving)
-                } else {
-                    context.showToast(R.string.snackbarDraftSaving)
-                }
-            }
-            DraftAction.SEND -> {
-                if (isTaskRoot) context.showToast(R.string.snackbarEmailSending)
-            }
+            DraftAction.SAVE -> context.showToast(R.string.snackbarDraftSaving)
+            DraftAction.SEND -> context.showToast(R.string.snackbarEmailSending)
```

Comment:
factorize the context.showToast

Reply:
Also, is it intended to change the logic of snackbar display ? 
Did it caused bug like 2 snackbar being displayed before ?

Reply:
We display toast only when really leaving the screen (ie when triggering the worker).